common-rest/src/main/java/org/uniprot/api/rest/service/BasicSearchService.java

```diff
@@ -100,13 +177,37 @@ public SolrRequest createSolrRequest(
         }
         requestBuilder.query(requestedQuery);
 
-        requestBuilder.addSort(solrSortClause.getSort(request.getSort(), hasScore));
+        if (solrSortClause != null) {
+            requestBuilder.addSort(solrSortClause.getSort(request.getSort(), hasScore));
+        }
 
-        if (request.hasFacets()) {
+        if (includeFacets && request.hasFacets()) {
             requestBuilder.facets(request.getFacetList());
             requestBuilder.facetConfig(facetConfig);
         }
 
+        // set the solr batch size if passed else use DEFAULT_SOLR_BATCH_SIZE(25)
+        requestBuilder.rows(this.solrBatchSize.orElse(DEFAULT_SOLR_BATCH_SIZE));
+
+        // if the requested size is less than batch size, set the batch size as requested size
+        Integer requestedSize = request.getSize();
+        if (isSizeLessOrEqualToSolrBatchSize(requestedSize)) {
+            requestBuilder.rows(requestedSize); // add the batch size for the solr query
+        }
+
+        if (requestedSize
+                == NumberUtils
```

Comment:
safer to use `.equals` rather than `==`, since `requestedSize` could be instantiated with a `new` operator in future, causing this check to always fail ...